### ISSUE TITLE
winget upgrade SURF.eduVPNClient never really worked

### ISSUE THREAD

**Issue by hueldoeu** [microsoft/winget-cli#4373]
### Brief description of your issue

winget upgrade SURF.eduVPNClient never really worked

https://github.com/Amebis/eduVPN/issues/241

### Steps to reproduce

cmd => winget upgrade SURF.eduVPNClient

### Expected behavior

cmd => winget upgrade SURF.eduVPNClient and eduVPN gets updates

### Actual behavior

cmd => winget upgrade SURF.eduVPNClient and an error appears

### Environment

```shell
Windows-Paket-Manager v1.7.10861
Copyright (c) Microsoft Corporation. Alle Rechte vorbehalten.

Windows: Windows.Desktop v10.0.19045.4291
Systemarchitektur: X64
Paket: Microsoft.DesktopAppInstaller v1.22.10861.0

WinGet-Verzeichnisse
-----------------------------------------------------------------------------------------------------------------------
Protokolle                                   %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\Local…
Benutzereinstellungen                        %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\Local…
Verzeichnis für portierbare Links (Benutzer) %LOCALAPPDATA%\Microsoft\WinGet\Links
Portables Linkverzeichnis (Computer)         C:\Program Files\WinGet\Links
Portierbarer Paketstamm (Benutzer)           %LOCALAPPDATA%\Microsoft\WinGet\Packages
Portierbarer Paketstamm                      C:\Program Files\WinGet\Packages
Portierbares Paketstamm (x86)                C:\Program Files (x86)\WinGet\Packages
Installationsprogrammdownloads               %USERPROFILE%\Downloads

Links
-----------------------------------------------------------------------------------------
Datenschutzerklärung              https://aka.ms/winget-privacy
Lizenzvereinbarung                https://aka.ms/winget-license
Hinweise von Drittanbietern       https://aka.ms/winget-3rdPartyNotice
Startseite                        https://aka.ms/winget
Windows Store-Nutzungsbedingungen https://www.microsoft.com/en-us/storedocs/terms-of-sale

Administratoreinstellung                  Status
-----------------------------------------------------
LocalManifestFiles                        Deaktiviert
BypassCertificatePinningForMicrosoftStore Deaktiviert
InstallerHashOverride                     Deaktiviert
LocalArchiveMalwareScanOverride           Deaktiviert
```

**Comment by Trenly** [comment]
Can you share the exact error you're getting? When I try, it upgrades just fine
```
PS C:\Users\WDAGUtilityAccount\Desktop\winget-pkgs> winget upgrade
Name                       Id                Version  Available
---------------------------------------------------------------
eduVPN Client 3.255.17 x64 SURF.eduVPNClient 3.255.17 3.255.18
1 upgrades available.
PS C:\Users\WDAGUtilityAccount\Desktop\winget-pkgs> winget upgrade SURF.eduVPNClient
Found eduVPN Client [SURF.eduVPNClient] Version 3.255.18
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://github.com/Amebis/eduVPN/releases/download/3.255.18/eduVPNClient_3.255.18_x64.msi
  ██████████████████████████████  11.5 MB / 11.5 MB
Successfully verified installer hash
Starting package install...
Successfully installed
PS C:\Users\WDAGUtilityAccount\Desktop\winget-pkgs> winget upgrade
No installed package found matching input criteria.
```

**Comment by hueldoeu** [comment]
Kein anwendbares Upgrade gefunden. Eine neuere Paketversion ist in einer konfigurierten Quelle verfügbar, aber sie gilt nicht für Ihr System oder Ihre Anforderungen.
https://github.com/Amebis/eduVPN/issues/241

**Comment by Trenly** [comment]
> No applicable upgrade found. A newer package version is available in a configured source, but it does not apply to your system or requirements. [Amebis/eduVPN#241](https://github.com/Amebis/eduVPN/issues/241)

Do you have a setting for preferring certain installer types in WinGet? Any Group Policies that could be affecting it?

The easiest way to diagnose the issue is if you can share the log file from `winget upgrade SURF.eduVPNClient --verbose --logs`

**Comment by hueldoeu** [comment]
my windows installation is a bit weird like i don't have 7-zip in the context menu and i had lots of problems with libreoffice templates (i solved it in the registry https://www.libreoffice-forum.de/viewtopic.php?f=2&t=38690&p=107122#p107122 ).  i also konw a laptop with windows 7, which can't get upgraded to windows 10 and it has a very strang GPO settings. i think i've also strange GPO (group policy objects)

i will wait for the next release and try cmd => winget upgrade SURF.eduVPNClient --verbose --logs

[WinGet-2024-04-13-00-43-05.515.log](https://github.com/microsoft/winget-cli/files/14964237/WinGet-2024-04-13-00-43-05.515.log)

**Comment by rozmansi** [comment]
I am not an expert in winget, so I may be completely wrong...

The end of your log file says:
> 2024-04-13 00:43:10.010 [CLI ] Search result size: 1
> 2024-04-13 00:43:10.011 [CLI ] Found one app. App id: SURF.eduVPNClient App name: eduVPN Client
> 2024-04-13 00:43:10.011 [CLI ] Market Filter created with market: DE
> 2024-04-13 00:43:10.011 [CLI ] Locale Comparator created with Required Locales: [] , Preferred Locales: [de-DE] , IsInstalledLocale: 1

SURF.eduVPNClient is authored as en-US. Hopefully, you using de-DE locale is not preventing winget from installing this MSI.

> 2024-04-13 00:43:10.011 [CLI ] Architecture Comparator created with allowed architectures: [X64, X86, Neutral]
> 2024-04-13 00:43:10.018 [REPO] Attempting to open pinning database: C:\Users\Nutzer\AppData\Local\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\pinning.db
> 2024-04-13 00:43:10.019 [REPO] ... opening existing pinning database
> 2024-04-13 00:43:10.019 [REPO] Opening database for Read at 'C:\Users\Nutzer\AppData\Local\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\pinning.db'
> 2024-04-13 00:43:10.019 [REPO] Opened Pinning Index with version [1.0], last write [2023-07-06 11:26:46.000]
> 2024-04-13 00:43:10.019 [REPO] Did not find pin Package=[{6CA364B8-497D-442A-B500-95C8765687D9}] Source=[*PredefinedInstalledSource]

It appears as if you have some specific version of the application _pinned_ in winget. (https://learn.microsoft.com/en-us/windows/package-manager/winget/pinning)

> 2024-04-13 00:43:10.032 [CLI ] Terminating context: 0x8a15002b at C:\__w\1\s\external\pkg\src\AppInstallerCLICore\Workflows\UpdateFlow.cpp:bc

**Comment by rozmansi** [comment]
Today, I have released a new version of eduVPN client app. Also on winget. Please report if the problem has been solved now.

**Comment by hueldoeu** [comment]
yes, worked very good https://github.com/Amebis/eduVPN/issues/241#issuecomment-2138011930

**Comment by hueldoeu** [comment]
7-zip is also finally back in the context menu (since version 24.05) andthe problems with libreoffice templates had been solved again (until next libreoffice update) https://www.libreoffice-forum.de/viewtopic.php?f=2&t=38690&p=107647#p107647
